### PR TITLE
[added] lower level tab components

### PR DESCRIPTION
**Pull request by jquense**
also deprecates horizontal tab positions in Tabs

### CODE REVIEW

File: docs/examples/.eslintrc [hunk 1]
```diff
@@ -58,6 +58,9 @@
     "SplitButton",
     "Tab",
     "Table",
+    "TabPane",
```
Comment (taion) [inline docs/examples/.eslintrc]: lol alphabet

File: docs/examples/LeftTabs.js [hunk 2]
```diff
@@ -1,9 +1,24 @@
 const tabsInstance = (
-  <Tabs defaultActiveKey={2} position="left" tabWidth={3}>
-    <Tab eventKey={1} title="Tab 1">Tab 1 content</Tab>
-    <Tab eventKey={2} title="Tab 2">Tab 2 content</Tab>
-    <Tab eventKey={3} title="Tab 3" disabled>Tab 3 content</Tab>
-  </Tabs>
+  <TabContainer id="left-tabs-example" defaultActiveKey="first">
```
Comment (taion) [inline docs/examples/LeftTabs.js]: Can this example use `<Row componentClass>` and `<Col componentClass>`?

File: docs/src/sections/TabsSection.js [hunk 3]
```diff
@@ -26,8 +26,17 @@ export default function TabsSection() {
       <p>Set the <code>animation</code> prop to <code>false</code></p>
       <ReactPlayground codeText={Samples.TabsNoAnimation} exampleClassName="bs-example-tabs" />
 
-      <h3><Anchor id="left-tabs">Left tabs</Anchor></h3>
-      <p>Set <code>position</code> to <code>"left"</code>. Optionally, <code>tabWidth</code> can be passed the number of columns for the tabs.</p>
+      <h3><Anchor id="left-tabs">Custom Tab layout</Anchor></h3>
```
Comment (taion) [inline docs/src/sections/TabsSection.js]: Probably worth changing this hash :stuck_out_tongue:

File: src/TabContainer.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+     * is specified.
+     */
+    id(props, ...args) {
+      let error = null;
+
+      if (!props.generateChildId) {
+        error = idPropType(props, ...args);
+
+        if (!error && !props.id) {
+          error = new Error('You must provide an either an `id` or a `generateChildId` prop to TabContainer');
```
Comment (taion) [inline src/TabContainer.js]: Is this a hard requirement now, or only for a11y? If the latter, might be worth noting.

File: docs/examples/LeftTabs.js (hunk 2 above)
Comment (jquense) [inline docs/examples/LeftTabs.js]: I can, I didn't know if maybe that was to much noise over the API. I guess the using the plain classes is also a bit confusing

File: src/TabContainer.js (hunk 4 above)
Comment (jquense) [inline src/TabContainer.js]: not a hard requirement, good to mention you are right.

File: docs/examples/LeftTabs.js (hunk 2 above)
Comment (taion) [inline docs/examples/LeftTabs.js]: Well, the clearfix is a bit confusing – the rest is just consistency.

File: src/Nav.js [hunk 5]
```diff
@@ -1,16 +1,38 @@
...
 
 import ValidComponentChildren from './utils/ValidComponentChildren';
-import createChainedFunction from './utils/createChainedFunction';
+import chain from './utils/createChainedFunction';
 
 import Collapse from './Collapse';
 
 class Nav extends React.Component {
 
+  componentDidUpdate() {
```
Comment (taion) [inline src/Nav.js]: I feel like a lot of this logic (and below) might be extractable into a separate `TabNav` component. Maybe expose something like `navItemProps` or `itemProps` on `Nav`, and pass down the tab props from above?

File: src/Nav.js [hunk 6]
```diff
@@ -77,19 +143,80 @@ class Nav extends React.Component {
...
+      'aria-controls': controls,
+      tabIndex: isActive ? tabIndex : -1,
+      onKeyDown: chain(
+        this.handleTabKeyDown.bind(this, onSelect || (()=>{})),
+        onKeyDown
+      )
+    };
+  }
+
+  handleTabKeyDown(onSelect, event) {
```
Comment (jquense) [inline src/Nav.js]: So this is a problem with moving the context stuff down to a TabNavItem, the focus behavior needs to be able to trigger `onSelect`, either this here _still_ needs to listen to tab container context, or some sort of ugly calling down to the child method.

Comment (taion) [inline src/Nav.js]: Oops. I guess this is the best way to go for now, then.

**Comment by jquense** [comment]
ready to go I think

**Pull request merged**